spec/api-web-frame-spec.js

```diff
@@ -138,4 +138,20 @@ describe('webFrame module', function () {
       webFrame.setLayoutZoomLevelLimits(0, 25)
     })
   })
+
+  it('calls a spellcheck provider', function (done) {
+    w = new BrowserWindow({show: false})
+    const misspelledWord = 'spleling'
+    w.webContents.on('did-finish-load', () => {
+      for (const char of misspelledWord) {
+        w.webContents.sendInputEvent({type: 'char', keyCode: char})
+      }
+      w.webContents.sendInputEvent({type: 'char', keyCode: ' '})
+    })
+    ipcMain.once('spec-spell-check', (e, text) => {
+      assert.equal(text, misspelledWord)
+      done()
+    })
+    w.loadURL(`file://${fixtures}/pages/webframe-spell-check.html`)
```

Comment:
We now have a couple of nice [helper functions](https://github.com/electron/electron/blob/master/spec/events-helpers.js) for event handling. They can help to simplify the test:

```js
it('calls a spellcheck provider', async () => {
  w = new BrowserWindow({show: false})
  w.loadURL(`file://${fixtures}/pages/webframe-spell-check.html`)
  await emittedOnce(w.webContents, 'did-finish-load')

  const spellcheckerFeedback = emittedOnce(ipcMain, 'spec-spell-check')
  const misspelledWord = 'spleling'
  for (const keyCode of [...misspelledWord, ' ']) {
     w.webContents.sendInputEvent({type: 'char', keyCode})
  }
  const [, text] = await spellcheckerFeedback

  expect(text).to.equal(misspelledWord)
}
```